ocpmodels/common/utils.py

```diff
@@ -322,11 +328,19 @@ def get_pbc_distances(
     distances = distance_vectors.norm(dim=-1)
 
     # redundancy: remove zero distances
-    nonzero_idx = torch.nonzero(distances).flatten()
+    nonzero_idx = torch.arange(len(distances))[distances != 0]
     edge_index = edge_index[:, nonzero_idx]
     distances = distances[nonzero_idx]
 
-    if return_offsets:
-        return edge_index, distances, offsets
+    if return_distance_vec:
+        distance_vec = distance_vectors[nonzero_idx]
+        if return_offsets:
+            return edge_index, distances, distance_vec, offsets
+        else:
+            return edge_index, distances, distance_vec
+
     else:
-        return edge_index, distances
+        if return_offsets:
+            return edge_index, distances, offsets
+        else:
+            return edge_index, distances
```

Comment:
We could return a dictionary instead which by default has `edge_index` and `distances`, and optionally `offsets` and `distance_vec` if those flags are `True`. That might be cleaner, but will need changes in all models.

Reply:
Something like the following maybe

```python
out = {
    "edge_index": edge_index,
    "distances": distances,
}

if return_offsets:
    out["offets] = offsets

if return_distance_vec:
    out["distance_vec] = distance_vectors[nonzero_idx]

return out
```